### PR TITLE
Added protection for breaking sugarcane stems

### PR DESCRIPTION
**Pull request by fsckmc**
Pretty simple change, but noticed the protection wasn't there for sugarcane

### CODE REVIEW

**Comment by fsckmc** [comment]
Sorry about the force push revert, forgot to use a feature branch, didn't think I'd be implementing more stuff, but the codebase is nice, and it's something I use every day. So figured I'd have a look at what I could try implementing.

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinPlayerControllerMP.java [hunk 1]
```diff
@@ -34,6 +34,15 @@ private void onPlayerDamageBlock(BlockPos loc, EnumFacing face, CallbackInfoRetu
             EntityPlayerSP p = mc.thePlayer;
             ItemStack heldItem = p.getHeldItem();
             Block block = mc.theWorld.getBlockState(loc).getBlock();
+            if (heldItem != null && block.equals(Blocks.reeds)) {
```
Comment (ViralTaco) [inline src/main/java/codes/biscuit/skyblockaddons/mixins/MixinPlayerControllerMP.java]: I suggest avoiding repeating code with this: 

```suggestion
        if (heldItem != null) {
        
        if (block.equals(Blocks.melon_stem) || block.equals(Blocks.pumpkin_stem)
        || (block.equals(Blocks.reeds) && mc.theWorld.getBlockState(loc.down()).getBlock() != Blocks.reeds) {
          
          cir.setReturnValue(false);
          if (System.currentTimeMillis() - lastMessage > 15000) {
            
            lastMessage = System.currentTimeMillis();
            
            main.getUtils().sendMessage(EnumChatFormatting.RED
              + Message.MESSAGE_CANCELLED_STEM_BREAK.getMessage());
          }
        }
      }
```
please disregard formatting

Comment (fsckmc) [reply src/main/java/codes/biscuit/skyblockaddons/mixins/MixinPlayerControllerMP.java]: Yeah, super basic change I should have caught, was quite late when I wrote this. my bad, thanks for finding!

**Comment by ViralTaco** [comment]
Wait, no. I meant… omg do I have to make a pull request now? 
I meant to replace the whole thing with the code snippet, you just added it to the code. 
(It's my fault for trying to github)

File: src/main/java/codes/biscuit/skyblockaddons/mixins/MixinPlayerControllerMP.java [hunk 2]
```diff
@@ -34,7 +34,20 @@ private void onPlayerDamageBlock(BlockPos loc, EnumFacing face, CallbackInfoRetu
...
+    
+    cir.setReturnValue(false);
+    if (System.currentTimeMillis() - lastMessage > 15000) {
+      
+      lastMessage = System.currentTimeMillis();
+      main.getUtils().sendMessage(EnumChatFormatting.RED
+        + Message.MESSAGE_CANCELLED_STEM_BREAK.getMessage());
+    }
+  }
+}
```
Comment (ViralTaco) [inline src/main/java/codes/biscuit/skyblockaddons/mixins/MixinPlayerControllerMP.java]: Anything below this line is redundant. I don't know how to use git hub… Please pm if you need the file, I'll just have to convert the formating

**Comment by fsckmc** [comment]
It's fine I'll fix it locally.

**Comment by fsckmc** [comment]
> Wait, no. I meant… omg do I have to make a pull request now?
> I meant to replace the whole thing with the code snippet, you just added it to the code.
> (It's my fault for trying to github)

It's fine, everyone sucks at github or git. Nobody really knows what they're doing :P

**Comment by fsckmc** [comment]
@biscuut btw sorry for submitting a PR without asking you, just wanted a project to work on

**Comment by biscuut** [comment]
thanks, i merged manually

**Pull request closed without merging**